python/looker_sdk/rtl/api_settings.py

```diff
@@ -32,13 +32,21 @@
 from looker_sdk.rtl import transport
 
 if sys.version_info >= (3, 8):
-    from typing import Protocol
+    from typing import Protocol, TypedDict
 else:
-    from typing_extensions import Protocol
+    from typing_extensions import Protocol, TypedDict
 
+from typing_extensions import Required
+
+class SettingsConfig(TypedDict, total=False):
+    client_id: Required[str]
+    client_secret: Required[str]
+    base_url: Required[str]
+    verify_ssl: str
+    timeout: str
 
 class PApiSettings(transport.PTransportSettings, Protocol):
-    def read_config(self) -> Dict[str, str]:
+    def read_config(self) -> SettingsConfig:
```

Comment:
I've added the type here and my editor was happy with it. Adding it to `read_config` would probably require further code changes, probably not worth it?

Reply:
let's see how the mypy and unittest runs go. If `TypeDict.Required` is available on all our supported python versions then it actually would be nice (actually probably required) to update the default implementation's type signature